### PR TITLE
feat(Tooltip): Change Tooltip to use the tippy library

### PR DESCRIPTION
**Pull request by Artikodin**
- [x] Feature
- [ ] Fix
- [ ] Enhancement

## Brief
After an issue was open on Sonar about the Tooltip which was half hidden due to the overflow scroll.
We decided to change the component Tooltip directly in Stardust project by adding a 3rd partie library called Tippy. [Doc](https://atomiks.github.io/tippyjs/)
We use the react wrapper for that library which, indeed, is more compatible to React. 
Check [tippy for react](https://github.com/atomiks/tippy.js-react).

[Issue](https://github.com/tillersystems/tiller-sonar/issues/583) referring to that PR. 

## How Has This Been Tested?

I have not tested this component since it depends on  [tippy for react](https://github.com/atomiks/tippy.js-react) who is well tested.

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/Tooltip/__tests__/Tooltip.test.js [hunk 1]
```diff
@@ -1,160 +1,16 @@
...
-
-  test('should toggle on mouseEnter/mouseLeave ', done => {
-    const buttonText = 'Show Tooltip';
-    const { queryByTestId, getByTestId, getByText } = render(
-      <Tooltip
-        hover
-        title="Ventes nettes (ventes brutes moins les réductions et les annulations) plus les taxes
-        sur la période séléctionnée."
-      >
+  test('should renders only the child element', () => {
```
Comment (sun-tea) [inline src/Tooltip/__tests__/Tooltip.test.js]: `should render`

File: src/Tooltip/README.md [hunk 2]
```diff
@@ -10,27 +10,29 @@ import { Tooltip } from '@tillersystems/stardust';
...
+All of the native [Tippy.js](https://atomiks.github.io/tippyjs/) options can be passed as props. Visit [All Options](https://atomiks.github.io/tippyjs/all-options/) to view the complete table.
+
+| propName     |     propType     |    defaultValue    | isRequired |
+| ------------ | :--------------: | :----------------: | :--------: |
+| `appearance` |     `string`     |       `dark`       |     -      |
+| `arrow`      |      `bool`      |      `false`       |     -      |
+| `boundary`   |     `string`     |   `scrollParent`   |     -      |
+| `children`   |      `node`      |       `null`       |     -      |
+| `className`  |     `string`     |        `''`        |     -      |
+| `content`    |     `string`     |        `''`        |     -      |
```
Comment (sun-tea) [inline src/Tooltip/README.md]: update propType of content according to what is defined on jsdoc of the component

File: src/Tooltip/index.js [hunk 3]
```diff
@@ -1,153 +1,153 @@
...
- * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment). It is the element where a tooltip is hooked to.
- * @param {string} width // Tooltip width.
  * @param {string} appearance // Appearance is used to set the color of the tooltip which can be dark or light.
- * @param {string} title // Tooltip main text.
+ * @param {boolean} arrow // Determines if an arrow should be added to the tooltip pointing toward the reference element.
+ * @param {boolean} boundary // The boundary preventOverflow modifier. Possible values:  "scrollParent",  "window", "viewport", or an HTMLElement.
+ * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment). It is the element where a tooltip is hooked to.
+ * @param {string} className // Class needed by styled components.
+ * @param {node, string, function} content // The content of the tooltip. Along with a string or element, you can use a function that takes the reference element as an argument and returns content.
+ * @param {string} maxWidth // Determines the maximum width of the tooltip.
```
Comment (sun-tea) [inline src/Tooltip/index.js]: The `README` says it can be string or number

File: src/Tooltip/index.js [hunk 4]
```diff
@@ -1,153 +1,153 @@
...
- * @param {bool} active // Boolean set if the Tooltip is showed or not.
- * @param {bool} hover // Boolean set to use click or hover to show the Tooltip.
- * @param {bool} top // Boolean set to positionate the tooltip above or below the element.
- * @param {string} arrowPositionX // Set position of the Tooltip's arrow.
- * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment). It is the element where a tooltip is hooked to.
- * @param {string} width // Tooltip width.
  * @param {string} appearance // Appearance is used to set the color of the tooltip which can be dark or light.
- * @param {string} title // Tooltip main text.
+ * @param {boolean} arrow // Determines if an arrow should be added to the tooltip pointing toward the reference element.
+ * @param {boolean} boundary // The boundary preventOverflow modifier. Possible values:  "scrollParent",  "window", "viewport", or an HTMLElement.
```
Comment (sun-tea) [inline src/Tooltip/index.js]: `{string}`

File: src/Tooltip/index.js [hunk 5]
```diff
@@ -1,153 +1,153 @@
...
+ * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment). It is the element where a tooltip is hooked to.
+ * @param {string} className // Class needed by styled components.
+ * @param {node, string, function} content // The content of the tooltip. Along with a string or element, you can use a function that takes the reference element as an argument and returns content.
+ * @param {string} maxWidth // Determines the maximum width of the tooltip.
+ * @param {string} placement // Positions the tooltip relative to its reference element.
```
Comment (sun-tea) [inline src/Tooltip/index.js]: `position the tooltip`

**Comment by sun-tea** [comment]
@spartDev I forcepushed just to fix the typo on commit title but I wasn't up-to-date with the remote 😬 sorry sorry. Can you forcepush your local branch?

**Comment by spartDev** [comment]
giphy (1)

**Pull request merged**